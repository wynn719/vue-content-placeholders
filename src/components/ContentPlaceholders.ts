import { computed, defineComponent, h } from 'vue-demi'
import '../styles.scss'

export default defineComponent({
  props: {
    rounded: {
      type: Boolean,
      default: false,
    },
    centered: {
      type: Boolean,
      default: false,
    },
    animated: {
      type: Boolean,
      default: true,
    },
  },
  setup(props, { slots }) {
    const classObject = computed(() => ({
      'vue-content-placeholders': true,
      'vue-content-placeholders-is-rounded': props.rounded,
      'vue-content-placeholders-is-centered': props.centered,
      'vue-content-placeholders-is-animated': props.animated,
    }))

    return () => slots.default ? h('div', { class: classObject.value }, slots.default()) : null
  },
})
