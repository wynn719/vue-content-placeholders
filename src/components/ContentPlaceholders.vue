<template>
  <div :class="classObject">
    <slot />
  </div>
</template>

<script lang="ts" setup>
import { computed } from 'vue'

const props = defineProps({
  rounded: {
    type: Boolean,
    default: false,
  },
  centered: {
    type: Boolean,
    default: false,
  },
  animated: {
    type: Boolean,
    default: true,
  },
})

const classObject = computed(() => ({
  'vue-content-placeholders': true,
  'vue-content-placeholders-is-rounded': props.rounded,
  'vue-content-placeholders-is-centered': props.centered,
  'vue-content-placeholders-is-animated': props.animated,
}))
</script>

<style lang="scss">
  @import '../styles.scss';
</style>
