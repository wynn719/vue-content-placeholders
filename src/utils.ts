export function usePrefix(prefix: string) {
  return `vue-content-placeholders-${prefix}` as const
}
