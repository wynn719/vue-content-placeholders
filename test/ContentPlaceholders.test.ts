import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import ContentPlaceholders from '@/components/ContentPlaceholders'
import { usePrefix } from '@/utils'

describe('ContentPlaceholders', () => {
  const createWrapper = (args?: any) => mount(ContentPlaceholders, {
    slots: {
      default: 'Default',
    },
    ...args,
  })

  it('should render correctly', () => {
    const wrapper = createWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should not render anything when slot is empty', () => {
    const wrapper = mount(ContentPlaceholders)
    expect(wrapper.html()).toBe('')
  })

  it('should support rounded', () => {
    const wrapper = createWrapper({
      props: {
        rounded: true,
      },
    })
    expect(wrapper.classes(usePrefix('is-rounded'))).toBe(true)
  })

  it('should support centered', () => {
    const wrapper = createWrapper({
      props: {
        centered: true,
      },
    })
    expect(wrapper.classes(usePrefix('is-centered'))).toBe(true)
  })

  it('should support animated', () => {
    const wrapper = createWrapper({
      props: {
        animated: true,
      },
    })
    expect(wrapper.classes(usePrefix('is-animated'))).toBe(true)
  })
})
