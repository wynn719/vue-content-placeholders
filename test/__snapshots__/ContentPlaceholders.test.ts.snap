// Vitest Snapshot v1

exports[`ContentPlaceholders > should render correctly 1`] = `"<div class=\\"vue-content-placeholders vue-content-placeholders-is-animated\\">Default</div>"`;
